src/pages/index.js

```diff
@@ -0,0 +1,257 @@
+      console.error(err);
+    });
+}
+
+// need an if/else statement that checks whether the card is currently liked or not.
+//hint: isLiked is true then we want to tell the server to remove a like
+// else we tell the server to remove the like
+// when finished, (inside .then) update view of heart button
+
+//New Card Popup
+const newCardPopup = new PopupWithForm(
+  "#add-card-modal",
+  handleAddCardFormSubmit
+);
+newCardPopup.setEventListeners();
+addNewCardButton.addEventListener("click", () => {
+  addCardFormValidator.resetValidation();
+  newCardPopup.openModal();
+});
+
+//Profile Edit Popup
+const newProfileEdit = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+newProfileEdit.setEventListeners();
+
+//Avatar Edit Popup
+const newAvatarEdit = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+avatarEditButton.addEventListener("click", () => {
+  newAvatarEdit.openModal();
+});
+const editAvatarValidator = new FormValidator(
+  validationSettings,
+  editProfileForm
+);
+
+/*-----------------------------------------------------------------*/
+/*                   "Handle" Functions                            */
+/*-----------------------------------------------------------------*/
+
+function handleProfileEditSubmit(inputValues) {
+  newProfileEdit.setSaving(true);
+  console.log(inputValues);
+  api.editUserInfo(inputValues).then(() => {
+    userInfo.setUserInfo({
+      name: inputValues.title,
+      about: inputValues.description,
+    });
+    newProfileEdit.setSaving(false);
+    newProfileEdit.closeModal();
+  });
+}
+
+function handleAvatarSubmit(inputValues) {
+  newAvatarEdit.setSaving(true);
+  api.updateAvatar(inputValues.link).then((user) => {
+    userInfo.setAvatarImg(user);
+    newAvatarEdit.setSaving(false);
+  });
+  newAvatarEdit.closeModal();
+}
+
+function handleAddCardFormSubmit(inputValues) {
+  newCardPopup.setSaving(true);
+  api.addNewCard(inputValues).then((card) => {
+    const cardEl = renderCard(card);
+    section.addItem(cardEl);
+    newCardPopup.closeModal();
+    newCardPopup.setSaving(false);
+  });
+}
+
+function handleDeleteClick(card) {
+  deletePopup.openModal();
+  deletePopup._setConfirmation(() => {
+    api.deleteCard(card._id).then(() => {
+      card.removeCard();
+      deletePopup.closeModal();
+    });
+  });
+}
+
+function handleCardClick(name, link) {
+  newImagePopup.openModal({ name, link });
+}
+
+avatarCloseButton.addEventListener("click", () => {
+  newAvatarEdit.closeModal();
+});
+
+/*-----------------------------------------------------------------*/
+/*                      Event Listeners                            */
+/*-----------------------------------------------------------------*/
+profileEditBtn.addEventListener("click", () => {
+  const userData = userInfo.getUserInfo();
+  profileTitleInput.value = userData.name.trim();
+  profileDescriptionInput.value = userData.description.trim();
+  newProfileEdit.openModal();
```

Comment:
**Could be improved**

Please, call `resetValidation` here to clear possible errors